src/System.ComponentModel.EventBasedAsync/tests/BackgroundWorkerTests.cs

```diff
@@ -0,0 +1,134 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.ComponentModel;
+using System.Threading;
+using System.Threading.Tasks;
+using Xunit;
+
+public class BackgroundWorkerTests
+{
+    private static ManualResetEventSlim manualResetEvent1 = new ManualResetEventSlim(false);
+    private static ManualResetEventSlim manualResetEvent2 = new ManualResetEventSlim(false);
+    private static ManualResetEventSlim manualResetEvent3 = new ManualResetEventSlim(false);
+
+    private static void Wait(int milliseconds)
+    {
+        System.Threading.Tasks.Task.Delay(milliseconds).Wait();
+    }
+
+    const int timeoutShort = 300;
+    const int timeoutLong = 3000;
+
+    [Fact]
+    public static void BackgroundWorkerWorks()
+    {
+        BackgroundWorker worker = new BackgroundWorker();
+
+        worker.DoWork += worker_DoWork;
+        worker.RunWorkerCompleted += worker_RunWorkerCompleted;
+        worker.ProgressChanged += worker_ProgressChanged;
+        worker.WorkerReportsProgress = true;
+
+        manualResetEvent1.Reset();
+        manualResetEvent2.Reset();
+
+        worker.RunWorkerAsync();
+
+        // Wait for signal from worker_ProgressChanged (which is called by worker_DoWork)
+        manualResetEvent1.Wait(timeoutLong);
```

Comment:
Are we using timeouts here because we're testing timeouts, or because we don't expect things to timeout but we don't want to hang in case there's a bug?  If it's the latter, 3 seconds is probably too short... it's feasible there could be a hiccup in the lab that would cause things to get bogged down for a few seconds.  We should also probably be asserting the result of Wait as true.